### PR TITLE
[TextFields] Renaming old default to legacy default.

### PR DESCRIPTION
**Pull request by willlarche**
Rename of files and opting out of new functionality.

No new functionality.

### CODE REVIEW

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m [hunk 1]
```diff
@@ -0,0 +1,1261 @@
...
+
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super init];
+  if (self) {
+    [self commonMDCTextInputControllerDefaultInitialization];
+
+    _activeColor =
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextInputControllerLegacyDefault.m:187]: If the decoder doesn't have a value for ActiveColorKey, won't this overwrite the default value we just set for _activeColor to nil?
Check for existence of keys?

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m [hunk 2]
```diff
@@ -0,0 +1,1261 @@
...
+}
+
+- (instancetype)initWithTextInput:(UIView<MDCTextInput> *)textInput {
+  self = [self init];
+  if (self) {
+    _textInput = textInput;
+  }
+
+  // This should happen last because it relies on the state of a ton of properties.
+  [self setupInput];
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextInputControllerLegacyDefault.m]: Consider moving inside if (self) block?
Usually I don't expect to see anything expect return self; after the if block.

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m [hunk 3]
```diff
@@ -0,0 +1,1261 @@
...
+@property(nonatomic, strong) UIFont *customTrailingFont;
+
+@end
+
+@implementation MDCTextInputControllerLegacyDefault
+
+@synthesize characterCounter = _characterCounter;
+@synthesize characterCountMax = _characterCountMax;
+@synthesize characterCountViewMode = _characterCountViewMode;
+@synthesize errorColor = _errorColor;
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextInputControllerLegacyDefault.m]: If you explicitly define setErrorColor and errorColor below do you need to @synthesize?
(Here and above / below)

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m [hunk 4]
```diff
@@ -0,0 +1,1261 @@
...
+- (void)dealloc {
+  [self unsubscribeFromNotifications];
+  [self unsubscribeFromKVO];
+}
+
+- (void)commonMDCTextInputControllerDefaultInitialization {
+  _characterCountViewMode = UITextFieldViewModeAlways;
+  _disabledColor = [[self class] disabledColorDefault];
+  _floatingEnabled = [[self class] isFloatingEnabledDefault];
+  _internalCharacterCounter = [MDCTextInputAllCharactersCounter new];
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextInputControllerLegacyDefault.m]: Internally, we prefer [[MDCClass alloc] init] over new.

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m [hunk 5]
```diff
@@ -0,0 +1,1261 @@
...
+    [self updatePlaceholderY];
+    [CATransaction commit];
+  } else {
+    [self updateLayout];
+  }
+
+  // Accessibility
+  if (self.textInput.isEditing && self.characterCountMax > 0) {
+    NSString *announcementString;
+    if (!announcementString.length) {
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextInputControllerLegacyDefault.m:1125]: announcementString was just created / set to nil on the line above.
Will announcementString.length ever not be 0?
Do we mean to test another string here?

**Review by ianegordon — CHANGES_REQUESTED** [review]
This is the first pass.  I'll take another when we've landed the PR this depends on so I can step through.

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m (hunk 3 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerLegacyDefault.m]: So true! There were a few others too.

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerLegacyDefault.m:187]: .activeColor actually handles nil in its setter and getting. It uses the default!

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerLegacyDefault.m]: Done!

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m (hunk 4 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerLegacyDefault.m]: Done!

File: components/TextFields/src/MDCTextInputControllerLegacyDefault.m (hunk 5 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerLegacyDefault.m:1125]: Good catch. This is from copying code where the state is different. Done.

**Comment by willlarche** [comment]
Ready for re-review.

**Review by ianegordon — APPROVED** [review]

**Pull request merged**